### ISSUE TITLE
salt:// transfers are very slow for large files

### ISSUE THREAD

**Issue by bradthurber** [saltstack/salt#25455]
Repro steps
-----------------
1) Place alargefile (100MB or so) on srv/salt on master
2) Run 
```
salt 'minionname' cp.get_url salt://alargefile /tmp/alargefile
```
3) watch /tmp/alargefile on the minion. For me, this transfer was going around 1MB a minute (very slow).

Versions
------------
```
[root@d-cdc-bthur-01 files]# salt-master --versions
           Salt: 2015.5.3
         Python: 2.6.6 (r266:84292, Nov 21 2013, 10:50:32)
         Jinja2: 2.2.1
       M2Crypto: 0.20.2
 msgpack-python: 0.4.6
   msgpack-pure: Not Installed
       pycrypto: 2.0.1
        libnacl: Not Installed
         PyYAML: 3.10
          ioflo: Not Installed
          PyZMQ: 14.5.0
           RAET: Not Installed
            ZMQ: 4.0.5
           Mako: 0.3.4
        Tornado: Not Installed
```

```
# salt-minion --versions
           Salt: 2015.5.3
         Python: 2.6.6 (r266:84292, Nov 21 2013, 10:50:32)
         Jinja2: 2.2.1
       M2Crypto: 0.20.2
 msgpack-python: 0.4.6
   msgpack-pure: Not Installed
       pycrypto: 2.0.1
        libnacl: Not Installed
         PyYAML: 3.10
          ioflo: Not Installed
          PyZMQ: 14.5.0
           RAET: Not Installed
            ZMQ: 4.0.5
           Mako: 0.3.4
        Tornado: Not Installed
```

I don't know how this may or may not be related to earlier issue #24048 - since apparently tornado is no longer required and I don't have it installed (if I believe salt --versions).

**Comment by jfindlay** [comment]
@bradthurber, thanks for the report.

**Comment by jfindlay** [comment]
@bradthurber, I have tried to reproduce this unsuccessfully.  Do you have any potential network issues?  Do the master or minion debug or trace logs show anything interesting?

**Comment by bradthurber** [comment]
@jfindlay thank you for trying to repro.  I just did another repro attempt with an even larger file and it worked fine (100MB in 5 secs - which is acceptable). 

Unfortunately, I think I simplified this test case a bit too much. My real use case is pulling large files out of a gitfs-hosted (pygit2/libgit2 backed) formula. I just tried that again and it is still slow. This probably deserves a different ticket #25672
